### ISSUE TITLE
False positive in DuplicateMethods

### ISSUE THREAD

**Issue by bquorning** [rubocop/rubocop#15028]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Lint/DuplicateMethods

### Steps to reproduce the problem

Run Rubocop on the following code:

```rb
stub_const('Foo',
           Module.new { def self.underscore(_); end })
stub_const('Bar',
           Module.new { def self.underscore(_); end })
```

This issue is very similar to the one reported in #15010.

### Expected behavior

The same method being defined on two different, unrelated anonymous Modules, should not be regarded as a duplicate.

### Actual behavior

```
Lint/DuplicateMethods: Method Object.underscore is defined at both file.rb:2 and file.rb:4.
           Module.new { def self.underscore(_); end })
                        ^^^^^^^^^^^^^^^^^^^
```

### RuboCop version

Any version after b59e6f74275501faf892d292bbb257009b5181e5 / #15019.

**Comment by bkuhlmann** [comment]
With the release of RuboCop 1.86.0, I'm seeing this crop up as a `Lint/DuplicateMethods` error for the following in my specs:

``` 
Module.new { def test(*one, **two, &block) = super one, two, yield(block) }
      .instance_method(:test)
      .parameters
```

This didn't happen prior to 1.86.0. I think this is another variant/edge case that should be allowed? I can open a formal issue if so.

**Comment by bquorning** [comment]
Yes, please open a new formal issue.

**Comment by Darhazer** [comment]
Wow I opened the Pandora box with https://github.com/rubocop/rubocop/pull/14996 😰
